### PR TITLE
Generalize marketplace thank you content

### PR DESCRIPTION
**Pull request by daledupreez**
#### Changes proposed in this Pull Request

* This PR refactors the code in the `MarketplaceThankYou` component added in #52655 such that the same file exposes a more general `ThankYou` component that supports the original use case as well as a new page we want to build for Professional Email (aka Titan).
* At present this is still an initial exploration, but it may be close to where we want to land for our first iteration.
* One area that may make sense for the support area is to allow for more specific text and/or links to be supplied, but that should be easy to take on separately.

#### Testing instructions

There are two things we need to test:
1. The original Yoast purchase flow, and
2. The visuals for the new Professional Email page

For (1), I would love to get some confirmation from @jdc91 that the testing approach from #52655 is still valid. I kicked the tires here by navigating to `/marketplace/thank-you/:siteSlug` for an Atomic site where I manually installed the Yoast plugin, and that seemed to work, but I have no idea if that's valid!

For (2), the visual work is incomplete for now, as I used the same image and didn't wire in all the relevant checks yet, but you can take a look at the visuals by accessing `/email/:domain/titan/thank-you/:siteSlug` on a site with a domain.

### CODE REVIEW

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+				/* TODO: Fix URL */
+				stepCta: (
+					<FullWidthButton href="https://titan.email">
+						{ translate( 'Get the app' ) }
+					</FullWidthButton>
+				),
+			},
+			{
+				stepKey: 'titan_whats_next_manage_email',
+				stepTitle: translate( 'Manage your email' ),
```
Comment (a8ci18n) [inline client/my-sites/email/titan-setup-thank-you/index.tsx:91]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **25** times:
`translate( 'Manage your email settings' )` *ES Score: 6*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/try/generalize-marketplace-thank-you)

**Comment by ddc22** [comment]
Hi @daledupreez I never imagined the new designs for the marketplace thank you page will start to be used in other areas so this component will be incredibly hard to refactor in a maintainable way without leaving confusing remnants. This start is great and will be more reusable for other variations of a thank you page. However, I personally think this approach of a newer generic version of a thank you page should be written separately from scratch. If anything feel free to move and/or duplicate, the already available styled UI components used inside the Yoast thank you page in a new generic location, so that it can be imported back into the Yoast. 

Basically just move all the code you require for the generic implementation of this thank you page into a seperate, suitable location with appropriate naming ( Eg: `/ThankYouPage2021`  or some name to indicate that it is a brand new variation of the existing thank you pages.) And also rename all components generically dropping the marketplace prefix from everything. Then the required components can be imported back into the Yoast thank you page. 

Other than that I think you are on the right path! :)
Let me know your thoughts about this.

**Comment by daledupreez** [comment]
Thanks for the comments, @jdc91!

I agree that we should move the Thank You content out more completely, but I wanted this PR to explore the general shape of the refactor and approach. I'll do more to separate out the pieces over the next few days, and I think that should help this move towards truly general components.

**Comment by AllTerrainDeveloper** [comment]
Hi! I improved the initial commit by moving the ThankYou component to a new calypso/component. I've checked that the original Yoast thank you page still works.

image

And Titan email thank you page is as below:
image

Also, I removed some ugly margins that were adding different colors in the border of the screens and added the icon for a more generic one.

We still need a valid URL for the download of the mobile apps.

File: client/components/thank-you/index.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+const MarketplaceThankYouHeader = styled.div`
+    width: 100%;
+    height: 240px;
+    background-color: var( --studio-gray-0 );
+    display: flex;
+    justify-content: center;
+}`;
+
+const MarketplaceThankyouSection = styled.div`
```
Comment (stephanethomas) [inline client/components/thank-you/index.tsx]: Super minor but this should probably be `MarketplaceThankYouSection` instead of `MarketplaceThankyouSection`.

**Comment by stephanethomas** [comment]
A heads-up that I've rebased that pull request off `trunk` as it was too old.

File: client/components/thank-you/index.tsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+	const { stepCta, stepDescription, stepKey, stepTitle } = props;
+
+	return (
+		<React.Fragment key={ stepKey }>
+			<h3>{ stepTitle }</h3>
+			<div>
+				<p>{ stepDescription }</p>
+				<div>{ stepCta }</div>
+			</div>
+		</React.Fragment>
```
Comment (stephanethomas) [inline client/components/thank-you/index.tsx:109]: Minor but we could use the `<></>` syntax here.

File: client/components/thank-you/index.tsx [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+							<MarketplaceHeaderTitle subtitle className="thank-you__body-header wp-brand-font">
+								{ translate( 'How can we help?' ) }
+							</MarketplaceHeaderTitle>
+							<p>
+								{ translate(
+									'Our Happiness Engineers are here if you need help, or if you have any questions.'
+								) }
+							</p>
+							<VerticalNav>
+								<VerticalNavItem path={ '/help/contact' }>
```
Comment (stephanethomas) [inline client/components/thank-you/index.tsx]: We should use the `CALYPSO_CONTACT` constant instead of hard coding that path.

File: client/components/thank-you/index.tsx [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+									{ translate( 'Ask a question' ) }
+								</VerticalNavItem>
+								<VerticalNavItem path={ localizeUrl( 'https://wordpress.com/support' ) }>
```
Comment (stephanethomas) [inline client/components/thank-you/index.tsx]: Similarly, we should use `SUPPORT_ROOT`.

File: client/components/thank-you/index.tsx [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+									{ translate( 'Support documentation' ) }
+								</VerticalNavItem>
+							</VerticalNav>
+						</MarketplaceThankyouSection>
+					) }
+				</div>
+			</ThankYouBody>
+		</MarketplaceThankYouContainer>
```
Comment (stephanethomas) [inline client/components/thank-you/index.tsx]: This file is a bit confusing. On one hand, it's named `index.tsx` which means it should refer to a generic `Thank You` page. One the other hand, it is filled with components named after `Marketplace`.

I'm also worried that this page lives in a new `client/components/thank-you/` folder whereas all previous `Thank You` pages are located in `client/my-sites/checkout/checkout-thank-you/`. Wouldn't it be better to keep all pages together (even if the current location is probably not great)?

File: client/my-sites/email/controller.js [hunk 7]
```diff
@@ -86,6 +87,17 @@ export default {
 		next();
 	},
 
+	emailManagementTitanSetupThankYouPage( pageContext, next ) {
```
Comment (stephanethomas) [inline client/my-sites/email/controller.js]: Should we really add this `Page` suffix whereas we don't use it in other controllers nor in the name of the page component? Moreover since this `Thank You` page is only for a specific mailbox I feel we could make that clear in the name, e.g. `emailManagementTitanSetupMailboxThankYou`. This may help in the future when we implement the flow where users can set up several unused mailboxes at once.

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+import { getTitanEmailUrl } from 'calypso/lib/titan';
+import Gridicon from 'calypso/components/gridicon';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+
+type TitanSetupThankYouProps = {
+	domainName: string;
+	emailAddress?: string;
+};
+
+export const TitanSetupThankYou = ( props: TitanSetupThankYouProps ): JSX.Element => {
```
Comment (stephanethomas) [inline client/my-sites/email/titan-setup-thank-you/index.tsx]: Should we use `ReactElement` instead of `JSX.Element` ?

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+	const currentRoute = useSelector( getCurrentRoute );
+	const selectedSiteSlug = useSelector( getSelectedSiteSlug );
+	const translate = useTranslate();
+	const thankYouImageSrc = '/calypso/images/upgrades/thank-you.svg';
```
Comment (stephanethomas) [inline client/my-sites/email/titan-setup-thank-you/index.tsx]: We could remove that variable, it's only used once.

Comment (stephanethomas) [reply client/my-sites/email/titan-setup-thank-you/index.tsx]: Also shouldn't the path be `calypso/assets/images/upgrades/thank-you.svg`?

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+	const { domainName, emailAddress = `youremail@${ domainName }` } = props;
+
+	const emailManagementPath = emailManagement( selectedSiteSlug, domainName, currentRoute );
+
+	const thankYouImage = {
+		alt: 'Thank you',
```
Comment (stephanethomas) [inline client/my-sites/email/titan-setup-thank-you/index.tsx]: This string should be localized.

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+				stepKey: 'titan_whats_next_get_mobile_app',
+				stepTitle: translate( 'Get mobile app' ),
+				stepDescription: translate(
+					"Access your email on the go with Titan's Android and iOS apps."
+				),
+				/* TODO: Fix URL: There are a some services that automatically redirects you to the Android or App Store
+				 * regarding your user's agent. Eg: https://tosto.re/ We might use a service like that, to take the user
+				 * to the store that is relative to hum (Windows Store? Apple Store? Android Store? Web app?*/
+				stepCta: (
+					<FullWidthButton href="https://titan.email">
```
Comment (stephanethomas) [inline client/my-sites/email/titan-setup-thank-you/index.tsx]: Aren't we supposed to redirect the user to the Titan Control Panel like we did [here](https://github.com/Automattic/wp-calypso/blob/9bee04a60fdd34d64c62b2591c599f3c0d4d6929/client/my-sites/email/email-management/titan-manage-mailboxes/index.jsx#L141)?

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+			},
+			{
+				stepKey: 'titan_whats_next_get_mobile_app',
+				stepTitle: translate( 'Get mobile app' ),
+				stepDescription: translate(
+					"Access your email on the go with Titan's Android and iOS apps."
+				),
+				/* TODO: Fix URL: There are a some services that automatically redirects you to the Android or App Store
+				 * regarding your user's agent. Eg: https://tosto.re/ We might use a service like that, to take the user
+				 * to the store that is relative to hum (Windows Store? Apple Store? Android Store? Web app?*/
```
Comment (stephanethomas) [inline client/my-sites/email/titan-setup-thank-you/index.tsx]: We should probably remove that `todo`.

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+				stepCta: (
+					<FullWidthButton href="https://titan.email">
+						{ translate( 'Get the app' ) }
```
Comment (stephanethomas) [inline client/my-sites/email/titan-setup-thank-you/index.tsx]: I feel the CTA should be `Get App`, and I would use `Get the mobile app` for the heading. That way all headings read almost like a sentence, and CTAs are consistent.

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+					</FullWidthButton>
+				),
+			},
+			{
+				stepKey: 'titan_whats_next_manage_email',
+				stepTitle: translate( 'Manage your email' ),
+				stepDescription: translate(
+					'Manage your emails, create new mailboxes, and import your emails, all within your dashboard.'
```
Comment (stephanethomas) [inline client/my-sites/email/titan-setup-thank-you/index.tsx]: There is a bit of redundancy between `Manage your emails` and `create new mailboxes`. I don't see how much value `all within your dashboard` brings; it is also incorrect as importing emails will happen in the Titan Control Panel, not the WordPress.com dashboard. As such, I'd like to propose the following:

> Add or delete mailboxes, migrate existing emails, configure a catch-all email, and much more.

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+	};
+
+	const titanThankYouSection = {
+		sectionKey: 'titan_whats_next',
+		sectionTitle: translate( 'What’s next?' ),
+		nextSteps: [
+			{
+				stepKey: 'titan_whats_next_view_inbox',
+				stepTitle: translate( 'Access your inbox' ),
+				stepDescription: translate( 'Access your email from anywhere with our built-in webmail.' ),
```
Comment (stephanethomas) [inline client/my-sites/email/titan-setup-thank-you/index.tsx]: I would remove `built-in` as we don't offer a webmail on WordPress.com yet.

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+	emailAddress?: string;
+};
+
+export const TitanSetupThankYou = ( props: TitanSetupThankYouProps ): JSX.Element => {
+	const currentRoute = useSelector( getCurrentRoute );
+	const selectedSiteSlug = useSelector( getSelectedSiteSlug );
+	const translate = useTranslate();
+	const thankYouImageSrc = '/calypso/images/upgrades/thank-you.svg';
+
+	const { domainName, emailAddress = `youremail@${ domainName }` } = props;
```
Comment (stephanethomas) [inline client/my-sites/email/titan-setup-thank-you/index.tsx]: I'm not a big fan of showing `youremail@` when the email address is not available, as it's not localized. We could simply show `Your email is now ready to use` in the header of the page when that's the case.

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+						{ translate( 'Go to Inbox' ) }
+						<Gridicon icon="external" />
+					</FullWidthButton>
+				),
+			},
+			{
+				stepKey: 'titan_whats_next_get_mobile_app',
+				stepTitle: translate( 'Get mobile app' ),
+				stepDescription: translate(
+					"Access your email on the go with Titan's Android and iOS apps."
```
Comment (stephanethomas) [inline client/my-sites/email/titan-setup-thank-you/index.tsx:80]: I was about to mention that we should refer to `Professional Email` instead of `Titan` but after giving it more thoughts this is probably one of the few places where `Titan` is a better option - users will download a `Titan` app after all. However if we keep it here, we should probably add it on the `Manage All Mailboxes` page then:

screenshot

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+		sectionTitle: translate( 'What’s next?' ),
+		nextSteps: [
+			{
+				stepKey: 'titan_whats_next_view_inbox',
+				stepTitle: translate( 'Access your inbox' ),
+				stepDescription: translate( 'Access your email from anywhere with our built-in webmail.' ),
+				stepCta: (
+					<FullWidthButton href={ getTitanEmailUrl( emailAddress ) } primary>
+						{ translate( 'Go to Inbox' ) }
+						<Gridicon icon="external" />
```
Comment (stephanethomas) [inline client/my-sites/email/titan-setup-thank-you/index.tsx]: We should fix the empty margin between the label and the icon:

screenshot

**Review by stephanethomas — CHANGES_REQUESTED** [review]
I left quite a few comments. I've also noticed a very subtle issue with scrolling -  when I scroll up or down I get block mid-course:

video

It feels like we have an invisible scrollbar for some elements of the page - so I start scrolling in that element and when I reach its end the page itself finally scrolls (see how the scrollbar behaves).

File: client/components/thank-you/index.tsx (hunk 3 above)
Comment (AllTerrainDeveloper) [reply client/components/thank-you/index.tsx:109]: Not fully sure about this is a possibility. We are adding a key prop to the React.Fragment here to avoid problems with React.

Comment (stephanethomas) [reply client/components/thank-you/index.tsx:109]: Oh, I've missed that part.

File: client/components/thank-you/index.tsx [hunk 19]
```diff
@@ -0,0 +1,174 @@
...
+	> div {
+		width: 460px;
+		padding: 0 35px;
+	}
+	div {
+		min-width: 144px;
+	}
+`;
+
+const MarketplaceNextSteps = styled.div< MarketplaceThemeProps >`
```
Comment (stephanethomas) [inline client/components/thank-you/index.tsx]: Shoudln't we rename `MarketplaceNextSteps` to `ThankYouNextSteps`?

File: client/components/thank-you/index.tsx [hunk 20]
```diff
@@ -0,0 +1,174 @@
...
+import { TranslateResult, useTranslate } from 'i18n-calypso';
+import React from 'react';
+import styled from '@emotion/styled';
+
+/**
+ * Internal dependencies
+ */
+import { CALYPSO_CONTACT, SUPPORT_ROOT } from 'calypso/lib/url/support';
+import { MarketplaceThemeProps } from 'calypso/my-sites/marketplace/theme';
+import { MarketplaceHeaderTitle } from 'calypso/my-sites/marketplace/components';
```
Comment (stephanethomas) [inline client/components/thank-you/index.tsx:15]: We are still referencing files from `calypso/my-sites/marketplace`, it would be great to find a way to address that. That would be for another pull request though.

File: client/my-sites/email/controller.js (hunk 7 above)
Comment (stephanethomas) [reply client/my-sites/email/controller.js]: This still has the `Page` suffix.

File: client/my-sites/email/controller.js [hunk 21]
```diff
@@ -86,6 +87,17 @@ export default {
...
+		pageContext.primary = (
+			<TitanSetupThankYou
+				domainName={ pageContext.params.domain }
+				emailAddress={ pageContext.query.email }
```
Comment (stephanethomas) [inline client/my-sites/email/controller.js:94]: The `TitanSetupThankYou` component is currently designed to retrieve the email address as a query parameter:

https://wordpress.com/email/:domain/titan/thank-you/:site?email=:email

Since `:email` should always point to the same domain than `:domain`, what if asked for the mailbox instead?

https://wordpress.com/email/:domain/titan/thank-you/:site?mailbox=:mailbox

We can indeed build the email address from these two pieces of information inside the `TitanSetupThankYou` component (i.e. `:mailbox@:domain`). We could even add that mailbox to the path so the url is even shorter:

https://wordpress.com/email/:domain/titan/thank-you/:mailbox/:site

That would give us something like:

https://wordpress.com/email/example.com/titan/thank-you/contact/example.wordpress.com

Comment (stephanethomas) [reply client/my-sites/email/controller.js:94]: Thinking about it more, since we will introduce the [new page](https://github.com/Automattic/wp-calypso/pull/54934) for setting up the unused mailbox at:

https://wordpress.com/email/:domain/titan/setup/:site

Or at - depending on the outcome of [this discussion](https://github.com/Automattic/wp-calypso/pull/54934#discussion_r678466357):

https://wordpress.com/email/:domain/titan/setup-mailbox/:site

It probably makes sense to move this new `Thank You` page at:

https://wordpress.com/email/:domain/titan/setup/thank-you/:mailbox/:site

File: client/my-sites/email/titan-setup-thank-you/index.tsx (hunk 10 above)
Comment (stephanethomas) [reply client/my-sites/email/titan-setup-thank-you/index.tsx]: This has not been addressed.

File: client/my-sites/email/titan-setup-thank-you/index.tsx (hunk 13 above)
Comment (stephanethomas) [reply client/my-sites/email/titan-setup-thank-you/index.tsx]: Looks like this also felt through the cracks.

File: client/my-sites/email/titan-setup-thank-you/index.tsx (hunk 17 above)
Comment (stephanethomas) [reply client/my-sites/email/titan-setup-thank-you/index.tsx:80]: The string in the `Manage All Mailboxes` page has not been updated, it should be:

> Download Titan's Android and iOS apps to access your emails on the go

File: client/my-sites/email/titan-setup-thank-you/index.tsx (hunk 15 above)
Comment (stephanethomas) [reply client/my-sites/email/titan-setup-thank-you/index.tsx]: This has not been addressed.

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 22]
```diff
@@ -0,0 +1,136 @@
...
+	const { domainName, emailAddress } = props;
+
+	const emailManagementPath = emailManagement( selectedSiteSlug, domainName, currentRoute );
+
+	const thankYouImage = {
+		alt: 'Thank you',
+		src: '/calypso/images/upgrades/thank-you.svg',
+	};
+
+	const getPath = ( context: string ) => {
```
Comment (stephanethomas) [inline client/my-sites/email/titan-setup-thank-you/index.tsx]: I wonder if `getControlPanelUrl` wouldn't be a better name for that function (it would better describe what it does).

**Review by stephanethomas — CHANGES_REQUESTED** [review]
Some of my previous feedback has not been addressed but this concerns minor changes. I've made some proposals regarding the route of the new `Thank You` page for Titan but we can wait for https://github.com/Automattic/wp-calypso/pull/54934 to be deployed before refining that. There is one blocker though, which is the display of these pages for Titan and Yoast which have either the top or the bottom cut:

screenshot

screenshot

File: client/my-sites/email/email-management/titan-manage-mailboxes/index.jsx [hunk 23]
```diff
@@ -134,7 +134,7 @@ class TitanManageMailboxes extends Component {
 
 						<VerticalNavItemEnhanced
 							description={ translate(
-								'Download our Android and iOS apps to access your emails on the go'
+								"Download Titan's Android and iOS apps to access your emails on the go"
```
Comment (a8ci18n) [inline client/my-sites/email/email-management/titan-manage-mailboxes/index.jsx]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 24]
```diff
@@ -0,0 +1,138 @@
...
+				stepKey: 'titan_whats_next_get_mobile_app',
+				stepTitle: translate( 'Get mobile app' ),
+				stepDescription: translate(
+					"Access your email on the go with Titan's Android and iOS apps."
+				),
+				stepCta: (
+					<FullWidthButton
+						href={ getControlPanelUrl( TITAN_CONTROL_PANEL_CONTEXT_GET_MOBILE_APP ) }
+					>
+						{ translate( 'Get app' ) }
```
Comment (a8ci18n) [inline client/my-sites/email/titan-setup-thank-you/index.tsx:84]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **66** times:
`translate( 'Get Apps' )` *ES Score: 7*

File: client/components/emails/email-signup-titan-card/index.jsx [hunk 25]
```diff
@@ -44,17 +44,13 @@ class EmailSignupTitanCard extends React.Component {
...
 
 		return (
 			<div className="email-signup-titan-card__suggestion-content">
 				<h3 className="email-signup-titan-card__title">
-					{ translate( 'youremail@%(domainName)s', {
-						args: { domainName: customDomainName },
-						comment:
-							'This is a sample email address for the user at their domain; %(domainName)s is a domain name, e.g. example.com',
-					} ) }
+					{ translate( 'Your email is now ready to use' ) }
```
Comment (daledupreez) [inline client/components/emails/email-signup-titan-card/index.jsx]: I think these wording changes should be moved to a different PR and discussed more broadly.

File: client/my-sites/email/email-management/titan-manage-mailboxes/index.jsx (hunk 23 above)
Comment (a8ci18n) [inline client/my-sites/email/email-management/titan-manage-mailboxes/index.jsx]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

File: client/my-sites/email/titan-setup-thank-you/index.tsx (hunk 24 above)
Comment (a8ci18n) [inline client/my-sites/email/titan-setup-thank-you/index.tsx:84]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **66** times:
`translate( 'Get Apps' )` *ES Score: 7*

File: client/components/thank-you/index.tsx [hunk 26]
```diff
@@ -0,0 +1,174 @@
+/**
+ * External dependencies
+ */
+import { TranslateResult, useTranslate } from 'i18n-calypso';
```
Comment (daledupreez) [inline client/components/thank-you/index.tsx]: Minor note: I think we can split this up so we're only importing the type for `TranslateResult` - I believe that means we only use the import during compilation and not at runtime.
```suggestion
import type { TranslateResult } from 'i18n-calypso';
import { useTranslate } from 'i18n-calypso';
```

File: client/my-sites/email/email-management/titan-manage-mailboxes/index.jsx (hunk 23 above)
Comment (daledupreez) [inline client/my-sites/email/email-management/titan-manage-mailboxes/index.jsx]: Can you move this to a separate PR? While we noticed it here, I don't think it's directly tied to the Thank You content.

File: client/my-sites/email/paths.js [hunk 27]
```diff
@@ -93,6 +93,21 @@ export function emailManagementTitanControlPanelRedirect(
...
+export function emailManagementTitanSetupThankYouPage(
+	siteName,
+	domainName,
+	emailAddress = null,
+	relativeTo = null
+) {
+	return emailManagementEdit(
+		siteName,
+		domainName,
+		'titan/thank-you',
```
Comment (daledupreez) [inline client/my-sites/email/paths.js]: I agree with @stephanethomas's [suggestion](https://github.com/Automattic/wp-calypso/pull/54577/files#r679143124) that we add this under `titan/setup/thank-you` instead.
```suggestion
		'titan/setup/thank-you',
```

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 28]
```diff
@@ -0,0 +1,138 @@
...
+ */
+import React from 'react';
+import { ThemeProvider } from 'emotion-theming';
+import { connect, useSelector } from 'react-redux';
+import { localize, useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { ThankYou } from 'calypso/components/thank-you';
```
Comment (daledupreez) [inline client/my-sites/email/titan-setup-thank-you/index.tsx]: Nit: these imports could be in alphabetical order. (Definitely not blocking.)

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 29]
```diff
@@ -0,0 +1,138 @@
...
+	const thankYouImage = {
+		alt: translate( 'Thank you' ),
+		src: '/calypso/images/upgrades/thank-you.svg',
+	};
+
+	const getControlPanelUrl = ( context: string ) => {
+		return emailManagementTitanControlPanelRedirect( selectedSiteSlug, domainName, currentRoute, {
+			context,
+		} );
+	};
```
Comment (daledupreez) [inline client/my-sites/email/titan-setup-thank-you/index.tsx]: Nit: It looks like we're only using this function from one location. So why not just set the redirect URL directly into a constant?
```suggestion
	const titanControlPanelUrl = emailManagementTitanControlPanelRedirect(
		selectedSiteSlug,
		domainName,
		currentRoute, {
			context: TITAN_CONTROL_PANEL_CONTEXT_GET_MOBILE_APP,
		}
	);
```

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 30]
```diff
@@ -0,0 +1,138 @@
...
+		],
+	};
+
+	return (
+		<ThemeProvider theme={ theme }>
+			<ThankYou
+				sections={ [ titanThankYouSection ] }
+				showSupportSection={ true }
+				thankYouImage={ thankYouImage }
+				thankYouTitle={ translate( 'Your email is now ready to use' ) }
```
Comment (daledupreez) [inline client/my-sites/email/titan-setup-thank-you/index.tsx:108]: It may be worth adding the email address here in a follow-up PR. We can either show `"Your email is now ready to use"` when we don't have the email address, or show something more specific when we do have the email address. But let's take that on in a follow-up so we can get this PR shipped.

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 31]
```diff
@@ -0,0 +1,138 @@
...
+import { TITAN_CONTROL_PANEL_CONTEXT_GET_MOBILE_APP } from 'calypso/lib/titan/constants';
+
+/**
+ * Import styles
+ */
+import './style.scss';
+
+type TitanSetupThankYouProps = {
+	domainName: string;
+	emailAddress: string;
```
Comment (daledupreez) [inline client/my-sites/email/titan-setup-thank-you/index.tsx]: I think we may want to make `emailAddress` nullable/optional - I think this formulation makes it required.

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 32]
```diff
@@ -0,0 +1,138 @@
...
+		selectedDomainName: ownProps.domainName,
+	} );
+
+	return {
+		currentRoute: getCurrentRoute( state ),
+		domain,
+		hasSubscription: hasEmailSubscription( domain ),
+		isLoadingPurchase:
+			isFetchingSitePurchases( state ) || ! hasLoadedSitePurchasesFromServer( state ),
+		purchase: getEmailPurchaseByDomain( state, domain ),
```
Comment (daledupreez) [inline client/my-sites/email/titan-setup-thank-you/index.tsx]: It doesn't look like we're using any of these props. I think we can remove them and the imported functions we're using to build them. It may be worth taking that on in a follow-up PR.
```suggestion
```

File: client/components/thank-you/style.scss [hunk 33]
```diff
@@ -0,0 +1,8 @@
+.layout__content {
+	padding-top: 0;
+	padding-bottom: 0;
+}
```
Comment (daledupreez) [inline client/components/thank-you/style.scss]: This can't be added as-is -- it will apply to `.layout_content` _everywhere_ in Calypso!

**Review by daledupreez — COMMENTED** [review]
I have some tweaks stemming from inspection, but I had continued scroll/cut-off content issues when testing, and those don't appear when accessing the existing Yoast thank you page via wpcalypso.

So I dived in and tried to clean up the CSS and some related things I noticed. There are still some smaller items, but I _think_ I simplified the really nasty scroll and scroll-bar issues that were making the component hard to work with, and I noticed a subtle issue where we'd affect the CSS for `.layout_content` across all pages!

I'll push up my fixes shortly, but I think we can wrap this up with a pairing session in the morning.

File: client/my-sites/email/email-management/titan-manage-mailboxes/index.jsx (hunk 23 above)
Comment (a8ci18n) [inline client/my-sites/email/email-management/titan-manage-mailboxes/index.jsx]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 34]
```diff
@@ -0,0 +1,126 @@
...
+				stepKey: 'titan_whats_next_get_mobile_app',
+				stepTitle: translate( 'Get mobile app' ),
+				stepDescription: translate(
+					"Access your email on the go with Titan's Android and iOS apps."
+				),
+				stepCta: (
+					<FullWidthButton
+						href={ getControlPanelUrl( TITAN_CONTROL_PANEL_CONTEXT_GET_MOBILE_APP ) }
+					>
+						{ translate( 'Get app' ) }
```
Comment (a8ci18n) [inline client/my-sites/email/titan-setup-thank-you/index.tsx:84]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **66** times:
`translate( 'Get Apps' )` *ES Score: 7*

File: client/components/emails/email-signup-titan-card/index.jsx (hunk 25 above)
Comment (daledupreez) [reply client/components/emails/email-signup-titan-card/index.jsx]: Dani took care of this before I submitted my comments!

File: client/my-sites/email/paths.js (hunk 27 above)
Comment (daledupreez) [reply client/my-sites/email/paths.js]: Actually, I take that back! Let's start with `titan/setup-mailbox/thank-you` to start.
```suggestion
		'titan/setup-mailbox/thank-you',
```

File: client/my-sites/email/titan-setup-thank-you/index.tsx (hunk 32 above)
Comment (daledupreez) [reply client/my-sites/email/titan-setup-thank-you/index.tsx]: I took care of this while I was digging into the CSS issues!

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 35]
```diff
@@ -0,0 +1,127 @@
...
+			},
+			{
+				stepKey: 'titan_whats_next_get_mobile_app',
+				stepTitle: translate( 'Get mobile app' ),
+				stepDescription: translate(
+					"Access your email on the go with Titan's Android and iOS apps."
+				),
+				stepCta: (
+					<FullWidthButton href={ titanControlPanelUrl }>
+						{ translate( 'Get app' ) }
```
Comment (a8ci18n) [inline client/my-sites/email/titan-setup-thank-you/index.tsx:84]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **66** times:
`translate( 'Get Apps' )` *ES Score: 7*

File: client/components/thank-you/style.scss (hunk 33 above)
Comment (daledupreez) [reply client/components/thank-you/style.scss]: I removed this.

File: client/my-sites/email/titan-setup-thank-you/index.tsx [hunk 36]
```diff
@@ -0,0 +1,128 @@
...
+			},
+			{
+				stepKey: 'titan_whats_next_get_mobile_app',
+				stepTitle: translate( 'Get mobile app' ),
+				stepDescription: translate(
+					"Access your email on the go with Titan's Android and iOS apps."
+				),
+				stepCta: (
+					<FullWidthButton href={ titanControlPanelUrl } target="_blank">
+						{ translate( 'Get app' ) }
```
Comment (a8ci18n) [inline client/my-sites/email/titan-setup-thank-you/index.tsx:84]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **66** times:
`translate( 'Get Apps' )` *ES Score: 7*

**Review by daledupreez — COMMENTED** [review]
Other than the small link issue I fixed in 36a34409acc06b258cbbf7c05452b2ee4556c441, this LGTM!

Can you just double-check my changes before we deploy this? (I also created this, so I can't approve the PR!)

**Review by AllTerrainDeveloper — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6305206

Hi @daledupreez, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Access your inbox`
- `Access your email from anywhere with our webmail.`
- `Go to Inbox`
- `Access your email on the go with Titan's Android and iOS apps.`
- `Get app`
- `Manage your email`
- `Add or delete mailboxes, migrate existing emails, configure a catch-all email, and much more.`
- `Your email is now ready to use`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6305206).
